src/reanimated2/WorkletEventHandler.ts

```diff
@@ -1,25 +1,39 @@
 'use strict';
-import type { __NativeEvent } from './commonTypes';
+import type { NativeSyntheticEvent } from 'react-native';
 import NativeReanimatedModule from './NativeReanimated';
 import { registerEventHandler, unregisterEventHandler } from './core';
+import type { EventPayload, ReanimatedEvent } from './hook/commonTypes';
 
-function jsListener<T extends __NativeEvent<T>>(
+type AssertNativeEvent<Event extends object> = NativeSyntheticEvent<
```

Comment:
I would recommend some name without "assert", because it reminds me this usage of generics:
```
type AssertNumber<T> = T extends number ? T : never;

const a = 42;
const b = 'Hello';

const assertedA: AssertNumber<typeof a> = a; //OK
const assertedB: AssertNumber<typeof b> = b; //ERROR
```

Reply:
What do you suggest? That's a tough nut to crack.